src/Components/WebAssembly/Server/src/DebugProxyLauncher.cs

```diff
@@ -141,21 +139,40 @@ private static void PassThroughConsoleOutput(Process process)
     private static void CompleteTaskWhenServerIsReady(Process aspNetProcess, TaskCompletionSource<string> taskCompletionSource)
     {
         string? capturedUrl = null;
+        var errorEncountered = false;
+
+        aspNetProcess.ErrorDataReceived += OnErrorDataReceived;
+        aspNetProcess.BeginErrorReadLine();
+
         aspNetProcess.OutputDataReceived += OnOutputDataReceived;
         aspNetProcess.BeginOutputReadLine();
 
+        void OnErrorDataReceived(object sender, DataReceivedEventArgs eventArgs)
+        {
+            if (!string.IsNullOrEmpty(eventArgs.Data))
+            {
+                taskCompletionSource.TrySetException(new InvalidOperationException(
+                    eventArgs.Data));
+                errorEncountered = true;
+            }
+        }
+
         void OnOutputDataReceived(object sender, DataReceivedEventArgs eventArgs)
         {
             if (string.IsNullOrEmpty(eventArgs.Data))
             {
-                taskCompletionSource.TrySetException(new InvalidOperationException(
-                    "No output has been recevied from the application."));
+                if (!errorEncountered)
+                {
+                    taskCompletionSource.TrySetException(new InvalidOperationException(
+                        "No output has been received from the application."));
```

Comment:
"no output" can be misleading like it wasm in the [original case](https://github.com/dotnet/runtime/issues/70841#issuecomment-1176254582). We are failing here because the expected output was not received.
Also, it would be useful to print all the lines received till this point, in case the proxy is stuck, or failed on something but didn't write to stderr.

Reply:
> Also, it would be useful to print all the lines received till this point, in case the proxy is stuck, or failed on something but didn't write to stderr.

I believe https://github.com/dotnet/aspnetcore/blob/7579e905b5f174da9fc9a18918324183b1fff8a3/src/Components/WebAssembly/Server/src/DebugProxyLauncher.cs#L135 would take care of that.

```suggestion
                        "Expected output has not been received from the application."));
```